react/src/views/Transfers/TransferShipmentsOverview/TransferShipmentsOverviewView.tsx

```diff
@@ -0,0 +1,171 @@
+import { useContext, useMemo } from "react";
+import { gql, useQuery } from "@apollo/client";
+import { Alert, AlertIcon, Button, Heading, Stack } from "@chakra-ui/react";
+import { Link } from "react-router-dom";
+import { GlobalPreferencesContext } from "providers/GlobalPreferencesProvider";
+import { SHIPMENT_FIELDS_FRAGMENT } from "queries/fragments";
+import { ShipmentsQuery } from "types/generated/graphql";
+import { AddIcon } from "@chakra-ui/icons";
+import { TableSkeleton } from "components/Skeletons";
+import { SelectColumnFilter } from "components/Table/Filter";
+import ShipmentsTable from "./components/ShipmentsTable";
+import { BaseOrgCell, BoxesCell, DirectionCell, StateCell } from "./components/TableCells";
+
+export const ALL_SHIPMENTS_QUERY = gql`
+  ${SHIPMENT_FIELDS_FRAGMENT}
+  query Shipments {
+    shipments {
+      ...ShipmentFields
+    }
+  }
+`;
+
+function TransferShipmentOverviewView() {
+  const { globalPreferences } = useContext(GlobalPreferencesContext);
+
+  // fetch shipments data
+  const { loading, error, data } = useQuery<ShipmentsQuery>(ALL_SHIPMENTS_QUERY);
+
+  // transform shipments data for UI
+  const graphqlToTableTransformer = (shipmentQueryResult: ShipmentsQuery | undefined) =>
+    shipmentQueryResult?.shipments.map((element) => {
+      if (globalPreferences?.availableBases && element?.sourceBase && element?.targetBase) {
```

Comment:
The PR that defines sourceBase and targetBase as non-null in the Shipment type is now in master, so I suppose you can remove these checks here

Reply:
done